File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-eq-shared-5-4/InterestLegCalculationPeriodDates.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class InterestLegCalculationPeriodDates : ISerialized
    {
        public InterestLegCalculationPeriodDates(XmlNode xmlNode)
        {
            XmlNode effectiveDateNode = xmlNode.SelectSingleNode("effectiveDate");
            
            if (effectiveDateNode != null)
            {
                if (effectiveDateNode.Attributes["href"] != null || effectiveDateNode.Attributes["id"] != null) 
                {
                    if (effectiveDateNode.Attributes["id"] != null) 
                    {
                        effectiveDateIDRef_ = effectiveDateNode.Attributes["id"].Value;
                        AdjustableOrRelativeDate ob = new AdjustableOrRelativeDate(effectiveDateNode);
                        IDManager.SetID(effectiveDateIDRef_, ob);
                    }
                    else if (effectiveDateNode.Attributes["href"] != null)
                    {
                        effectiveDateIDRef_ = effectiveDateNode.Attributes["href"].Value;
                    }
                    else
                    {
                        effectiveDate_ = new AdjustableOrRelativeDate(effectiveDateNode);
                    }
                }
                else
                {
                    effectiveDate_ = new AdjustableOrRelativeDate(effectiveDateNode);
                }
            }
            
        
            XmlNode terminationDateNode = xmlNode.SelectSingleNode("terminationDate");
            
            if (terminationDateNode != null)
            {
                if (terminationDateNode.Attributes["href"] != null || terminationDateNode.Attributes["id"] != null) 
                {
                    if (terminationDateNode.Attributes["id"] != null) 
                    {
                        terminationDateIDRef_ = terminationDateNode.Attributes["id"].Value;
                        AdjustableOrRelativeDate ob = new AdjustableOrRelativeDate(terminationDateNode);
                        IDManager.SetID(terminationDateIDRef_, ob);
                    }
                    else if (terminationDateNode.Attributes["href"] != null)
                    {
                        terminationDateIDRef_ = terminationDateNode.Attributes["href"].Value;
                    }
                    else
                    {
                        terminationDate_ = new AdjustableOrRelativeDate(terminationDateNode);
                    }
                }
                else
                {
                    terminationDate_ = new AdjustableOrRelativeDate(terminationDateNode);
                }
            }
            
        
            XmlNode interestLegResetDatesNode = xmlNode.SelectSingleNode("interestLegResetDates");
            
            if (interestLegResetDatesNode != null)
            {
                if (interestLegResetDatesNode.Attributes["href"] != null || interestLegResetDatesNode.Attributes["id"] != null) 
                {
                    if (interestLegResetDatesNode.Attributes["id"] != null) 
                    {
                        interestLegResetDatesIDRef_ = interestLegResetDatesNode.Attributes["id"].Value;
                        InterestLegResetDates ob = new InterestLegResetDates(interestLegResetDatesNode);
                        IDManager.SetID(interestLegResetDatesIDRef_, ob);
                    }
                    else if (interestLegResetDatesNode.Attributes["href"] != null)
                    {
                        interestLegResetDatesIDRef_ = interestLegResetDatesNode.Attributes["href"].Value;
                    }
                    else
                    {
                        interestLegResetDates_ = new InterestLegResetDates(interestLegResetDatesNode);
                    }
                }
                else
                {
                    interestLegResetDates_ = new InterestLegResetDates(interestLegResetDatesNode);
                }
            }
            
        
            XmlNode interestLegPaymentDatesNode = xmlNode.SelectSingleNode("interestLegPaymentDates");
            
            if (interestLegPaymentDatesNode != null)
            {
                if (interestLegPaymentDatesNode.Attributes["href"] != null || interestLegPaymentDatesNode.Attributes["id"] != null) 
                {
                    if (interestLegPaymentDatesNode.Attributes["id"] != null) 
                    {
                        interestLegPaymentDatesIDRef_ = interestLegPaymentDatesNode.Attributes["id"].Value;
                        AdjustableRelativeOrPeriodicDates2 ob = new AdjustableRelativeOrPeriodicDates2(interestLegPaymentDatesNode);
                        IDManager.SetID(interestLegPaymentDatesIDRef_, ob);
                    }
                    else if (interestLegPaymentDatesNode.Attributes["href"] != null)
                    {
                        interestLegPaymentDatesIDRef_ = interestLegPaymentDatesNode.Attributes["href"].Value;
                    }
                    else
                    {
                        interestLegPaymentDates_ = new AdjustableRelativeOrPeriodicDates2(interestLegPaymentDatesNode);
                    }
                }
                else
                {
                    interestLegPaymentDates_ = new AdjustableRelativeOrPeriodicDates2(interestLegPaymentDatesNode);
                }
            }
            
        
        }
        
    
        #region EffectiveDate_
        private AdjustableOrRelativeDate effectiveDate_;
        public AdjustableOrRelativeDate EffectiveDate_
        {
            get
            {
                if (this.effectiveDate_ != null)
                {
                    return this.effectiveDate_; 
                }
                else if (this.effectiveDateIDRef_ != null)
                {
                    effectiveDate_ = IDManager.getID(effectiveDateIDRef_) as AdjustableOrRelativeDate;
                    return this.effectiveDate_; 
                }
                else
                {
                      return this.effectiveDate_; 
                }
            }
            set
            {
                if (this.effectiveDate_ != value)
                {
                    this.effectiveDate_ = value;
                }
            }
        }
        #endregion
        
        public string effectiveDateIDRef_ { get; set; }
        #region TerminationDate_
        private AdjustableOrRelativeDate terminationDate_;
        public AdjustableOrRelativeDate TerminationDate_
        {
            get
            {
                if (this.terminationDate_ != null)
                {
                    return this.terminationDate_; 
                }
                else if (this.terminationDateIDRef_ != null)
                {
                    terminationDate_ = IDManager.getID(terminationDateIDRef_) as AdjustableOrRelativeDate;
                    return this.terminationDate_; 
                }
                else
                {
                      return this.terminationDate_; 
                }
            }
            set
            {
                if (this.terminationDate_ != value)
                {
                    this.terminationDate_ = value;
                }
            }
        }
        #endregion
        
        public string terminationDateIDRef_ { get; set; }
        #region InterestLegResetDates_
        private InterestLegResetDates interestLegResetDates_;
        public InterestLegResetDates InterestLegResetDates_
        {
            get
            {
                if (this.interestLegResetDates_ != null)
                {
                    return this.interestLegResetDates_; 
                }
                else if (this.interestLegResetDatesIDRef_ != null)
                {
                    interestLegResetDates_ = IDManager.getID(interestLegResetDatesIDRef_) as InterestLegResetDates;
                    return this.interestLegResetDates_; 
                }
                else
                {
                      return this.interestLegResetDates_; 
                }
            }
            set
            {
                if (this.interestLegResetDates_ != value)
                {
                    this.interestLegResetDates_ = value;
                }
            }
        }
        #endregion
        
        public string interestLegResetDatesIDRef_ { get; set; }
        #region InterestLegPaymentDates_
        private AdjustableRelativeOrPeriodicDates2 interestLegPaymentDates_;
        public AdjustableRelativeOrPeriodicDates2 InterestLegPaymentDates_
        {
            get
            {
                if (this.interestLegPaymentDates_ != null)
                {
                    return this.interestLegPaymentDates_; 
                }
                else if (this.interestLegPaymentDatesIDRef_ != null)
                {
                    interestLegPaymentDates_ = IDManager.getID(interestLegPaymentDatesIDRef_) as AdjustableRelativeOrPeriodicDates2;
                    return this.interestLegPaymentDates_; 
                }
                else
                {
                      return this.interestLegPaymentDates_; 
                }
            }
            set
            {
                if (this.interestLegPaymentDates_ != value)
                {
                    this.interestLegPaymentDates_ = value;
                }
            }
        }
        #endregion
        
        public string interestLegPaymentDatesIDRef_ { get; set; }
        
    
        
    
    }
    
}

File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-eq-shared-5-4/InterestLegCalculationPeriodDates.cpp
// InterestLegCalculationPeriodDates.cpp 
#include "InterestLegCalculationPeriodDates.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

InterestLegCalculationPeriodDates::InterestLegCalculationPeriodDates(TiXmlNode* xmlNode)
: ISerialized(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //effectiveDateNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* effectiveDateNode = xmlNode->FirstChildElement("effectiveDate");

   if(effectiveDateNode){effectiveDateIsNull_ = false;}
   else{effectiveDateIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- effectiveDateNode , address : " << effectiveDateNode << std::endl;
   #endif
   if(effectiveDateNode)
   {
      if(effectiveDateNode->Attribute("href") || effectiveDateNode->Attribute("id"))
      {
          if(effectiveDateNode->Attribute("id"))
          {
             effectiveDateIDRef_ = effectiveDateNode->Attribute("id");
             effectiveDate_ = boost::shared_ptr<AdjustableOrRelativeDate>(new AdjustableOrRelativeDate(effectiveDateNode));
             effectiveDate_->setID(effectiveDateIDRef_);
             IDManager::instance().setID(effectiveDateIDRef_,effectiveDate_);
          }
          else if(effectiveDateNode->Attribute("href")) { effectiveDateIDRef_ = effectiveDateNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { effectiveDate_ = boost::shared_ptr<AdjustableOrRelativeDate>(new AdjustableOrRelativeDate(effectiveDateNode));}
   }

   //terminationDateNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* terminationDateNode = xmlNode->FirstChildElement("terminationDate");

   if(terminationDateNode){terminationDateIsNull_ = false;}
   else{terminationDateIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- terminationDateNode , address : " << terminationDateNode << std::endl;
   #endif
   if(terminationDateNode)
   {
      if(terminationDateNode->Attribute("href") || terminationDateNode->Attribute("id"))
      {
          if(terminationDateNode->Attribute("id"))
          {
             terminationDateIDRef_ = terminationDateNode->Attribute("id");
             terminationDate_ = boost::shared_ptr<AdjustableOrRelativeDate>(new AdjustableOrRelativeDate(terminationDateNode));
             terminationDate_->setID(terminationDateIDRef_);
             IDManager::instance().setID(terminationDateIDRef_,terminationDate_);
          }
          else if(terminationDateNode->Attribute("href")) { terminationDateIDRef_ = terminationDateNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { terminationDate_ = boost::shared_ptr<AdjustableOrRelativeDate>(new AdjustableOrRelativeDate(terminationDateNode));}
   }

   //interestLegResetDatesNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* interestLegResetDatesNode = xmlNode->FirstChildElement("interestLegResetDates");

   if(interestLegResetDatesNode){interestLegResetDatesIsNull_ = false;}
   else{interestLegResetDatesIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- interestLegResetDatesNode , address : " << interestLegResetDatesNode << std::endl;
   #endif
   if(interestLegResetDatesNode)
   {
      if(interestLegResetDatesNode->Attribute("href") || interestLegResetDatesNode->Attribute("id"))
      {
          if(interestLegResetDatesNode->Attribute("id"))
          {
             interestLegResetDatesIDRef_ = interestLegResetDatesNode->Attribute("id");
             interestLegResetDates_ = boost::shared_ptr<InterestLegResetDates>(new InterestLegResetDates(interestLegResetDatesNode));
             interestLegResetDates_->setID(interestLegResetDatesIDRef_);
             IDManager::instance().setID(interestLegResetDatesIDRef_,interestLegResetDates_);
          }
          else if(interestLegResetDatesNode->Attribute("href")) { interestLegResetDatesIDRef_ = interestLegResetDatesNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { interestLegResetDates_ = boost::shared_ptr<InterestLegResetDates>(new InterestLegResetDates(interestLegResetDatesNode));}
   }

   //interestLegPaymentDatesNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* interestLegPaymentDatesNode = xmlNode->FirstChildElement("interestLegPaymentDates");

   if(interestLegPaymentDatesNode){interestLegPaymentDatesIsNull_ = false;}
   else{interestLegPaymentDatesIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- interestLegPaymentDatesNode , address : " << interestLegPaymentDatesNode << std::endl;
   #endif
   if(interestLegPaymentDatesNode)
   {
      if(interestLegPaymentDatesNode->Attribute("href") || interestLegPaymentDatesNode->Attribute("id"))
      {
          if(interestLegPaymentDatesNode->Attribute("id"))
          {
             interestLegPaymentDatesIDRef_ = interestLegPaymentDatesNode->Attribute("id");
             interestLegPaymentDates_ = boost::shared_ptr<AdjustableRelativeOrPeriodicDates2>(new AdjustableRelativeOrPeriodicDates2(interestLegPaymentDatesNode));
             interestLegPaymentDates_->setID(interestLegPaymentDatesIDRef_);
             IDManager::instance().setID(interestLegPaymentDatesIDRef_,interestLegPaymentDates_);
          }
          else if(interestLegPaymentDatesNode->Attribute("href")) { interestLegPaymentDatesIDRef_ = interestLegPaymentDatesNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { interestLegPaymentDates_ = boost::shared_ptr<AdjustableRelativeOrPeriodicDates2>(new AdjustableRelativeOrPeriodicDates2(interestLegPaymentDatesNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<AdjustableOrRelativeDate> InterestLegCalculationPeriodDates::getEffectiveDate()
{
   if(!this->effectiveDateIsNull_){
        if(effectiveDateIDRef_ != ""){
             return boost::shared_static_cast<AdjustableOrRelativeDate>(IDManager::instance().getID(effectiveDateIDRef_));
        }else{
             return this->effectiveDate_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<AdjustableOrRelativeDate>();
   }
}
boost::shared_ptr<AdjustableOrRelativeDate> InterestLegCalculationPeriodDates::getTerminationDate()
{
   if(!this->terminationDateIsNull_){
        if(terminationDateIDRef_ != ""){
             return boost::shared_static_cast<AdjustableOrRelativeDate>(IDManager::instance().getID(terminationDateIDRef_));
        }else{
             return this->terminationDate_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<AdjustableOrRelativeDate>();
   }
}
boost::shared_ptr<InterestLegResetDates> InterestLegCalculationPeriodDates::getInterestLegResetDates()
{
   if(!this->interestLegResetDatesIsNull_){
        if(interestLegResetDatesIDRef_ != ""){
             return boost::shared_static_cast<InterestLegResetDates>(IDManager::instance().getID(interestLegResetDatesIDRef_));
        }else{
             return this->interestLegResetDates_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<InterestLegResetDates>();
   }
}
boost::shared_ptr<AdjustableRelativeOrPeriodicDates2> InterestLegCalculationPeriodDates::getInterestLegPaymentDates()
{
   if(!this->interestLegPaymentDatesIsNull_){
        if(interestLegPaymentDatesIDRef_ != ""){
             return boost::shared_static_cast<AdjustableRelativeOrPeriodicDates2>(IDManager::instance().getID(interestLegPaymentDatesIDRef_));
        }else{
             return this->interestLegPaymentDates_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<AdjustableRelativeOrPeriodicDates2>();
   }
}
}

